CMakeLists.txt

```diff
@@ -11,6 +11,10 @@ set(CMAKE_C_STANDARD_REQUIRED ON)
 set(LPAC_CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 list(APPEND CMAKE_MODULE_PATH ${LPAC_CMAKE_MODULE_PATH})
 
+if(CMAKE_EXPORT_COMPILE_COMMANDS STREQUAL "")
```

Comment:
Please use `NOT DEFINED` instead.

Reply:
Do you mean something like that?

```
if(NOT DEFINED CMAKE_EXPORT_COMPILE_COMMANDS)
    set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
endif()
```

This doesn't work.

Reply:
Or we can omit the guard completely and force compile_commands.json creation. Meson build system makes this by default. It also creates a .gitignore file inside the build directory.

Reply:
@CoelacanthusHex

Reply:
I think it's better to follow Meson to remove guard. `.gitignore` is unnecessary because out-of-source build is recommended.

Reply:
OK. Than I'll send an update

Reply:
Ready

Reply:
@CoelacanthusHex `.gitignore` in the out-of-source folder is really nice. So far, we have `build` entry in the root `.gitignore`. But if you need another build folder `build-cross`, it will immidiatelly "pollute" you git repo. This is why meson automatically creates a `.gitignore` file inside the out-of-source build directory. Because in this case, git just ignores it without an entry inside the root `.gitignore` file. See this [post](https://www.scivision.dev/cmake-auto-gitignore-build-dir/).